libs/string_util/include/hpx/string_util/classification.hpp

```diff
@@ -0,0 +1,48 @@
+//  Copyright (c) 2020 ETH Zurich
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef HPX_STRING_UTIL_CLASSIFICATION_HPP
+#define HPX_STRING_UTIL_CLASSIFICATION_HPP
+
+#include <cctype>
+#include <string>
+
+namespace hpx { namespace string_util {
+    namespace detail {
+        template <typename CharT>
+        struct is_any_of_pred
+        {
+            bool operator()(int c)
+            {
+                return chars.find(c) != std::string::npos;
+            }
+
+            std::basic_string<CharT> chars;
+        };
+    }    // namespace detail
+
+    template <typename CharT>
+    detail::is_any_of_pred<CharT> is_any_of(
+        std::basic_string<CharT> const& chars)
+    {
+        return detail::is_any_of_pred{chars};
```

Comment:
This relies on CTAD (C++17), could you explicitly provide the `<CharT>` argument, please?

Reply:
Good catch, will fix.